python/arcticdb/storage_fixtures/s3.py

```diff
@@ -217,6 +243,241 @@ def real_s3_from_environment_variables(*, shared_path: bool):
+    except iam_client.exceptions.EntityAlreadyExistsException:
+        out.aws_policy_name = f"arn:aws:iam::{account_id}:policy/{policy_name}"
+        print("Policy already exists.")
+    except Exception as e:
+        print(f"Error creating policy: {e}")
+        raise e
+
+    # Attach the policy to the role
+    try:
+        iam_client.attach_role_policy(
+            RoleName=role_name,
+            PolicyArn=out.aws_policy_name
+        )
+        print("Policyattached to role successfully.")
+    except Exception as e:
+        print(f"Error attaching policy to role: {e}")
+        raise e
+
+    # Create an inline policy for the user to assume the role
+    assume_role_user_policy_document = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Resource": f"arn:aws:iam::{account_id}:role/{role_name}",
+            }
+        ]
+    }
+
+    try:
+        iam_client.put_user_policy(
+            UserName=user_name,
+            PolicyName="AssumeRolePolicy",
+            PolicyDocument=json.dumps(assume_role_user_policy_document)
+        )
+        print("Inline policy to assume role attached to user successfully.")
+    except Exception as e:
+        print(f"Error attaching inline policy to user: {e}")
+        raise e
+
+    print("User created with role to access bucket.")
+
+    try:
+        access_key_response = iam_client.create_access_key(UserName=user_name)
+        out.sts_test_key.id = access_key_response["AccessKey"]["AccessKeyId"]
+        out.sts_test_key.secret = access_key_response["AccessKey"]["SecretAccessKey"]
+        print("Access key created successfully.")
+    except Exception as e:
+        print(f"Error creating access key: {e}")
+        raise e
+    
+    out.aws_auth = AWSAuthMethod.STS_PROFILE_CREDENTIALS_PROVIDER
+    out.aws_profile = profile_name
+    real_s3_sts_write_local_credentials(out)
+    return out
+
+
+def real_s3_sts_write_local_credentials(factory: BaseS3StorageFixtureFactory):
+    base_profile_name = factory.aws_profile + "_base"
+    aws_credentials = f"""
+[profile {factory.aws_profile}]
+role_arn = {factory.aws_role_arn}
+source_profile = {base_profile_name}
+
+[profile {base_profile_name}]
+aws_access_key_id = {factory.sts_test_key.id}
+aws_secret_access_key = {factory.sts_test_key.secret}
+"""
+    aws_dir = os.path.dirname(S3_CONFIG_PATH)
+    if not os.path.exists(aws_dir):
+        os.makedirs(aws_dir)
+
+    if os.path.exists(S3_CONFIG_PATH):
+        os.rename(S3_CONFIG_PATH, S3_BACKUP_CONFIG_PATH)
+
+    with open(S3_CONFIG_PATH, "w") as config_file:
+        config_file.write(aws_credentials)
+
+
+def real_s3_sts_resources_ready(factory: BaseS3StorageFixtureFactory): 
+    sts_client = boto3.client(
+        "sts",
+        aws_access_key_id=factory.sts_test_key.id,
+        aws_secret_access_key=factory.sts_test_key.secret
+    )
+    for _ in range(20):
```

Comment:
20 retries seems excessive?

Reply:
Once it requires 13 seconds to get the resources available in S3